session.go

```diff
@@ -109,7 +109,8 @@ type session struct {
 
 	sessionCreationTime     time.Time
 	lastNetworkActivityTime time.Time
-	remoteIdleTimeout       time.Duration
+
+	peerParams *handshake.TransportParameters
```

Comment:
Can you explain this change? I'd prefer having the params as normal members instead of in the struct: Lesser coupling and simpler to reason about.

Reply:
`newStream` is checking if the `peerParams` are set, and using the value to initialize the flow controller.

Reply:
Couldn’t you do the same check using the params themselves?

Reply:
I don't understand where you're going with this. Of course we could create 3 or 4 new members of the `session` here, but I don't see any benefit in doing so (since they're all set at the same time). I'll merge this PR now, we can still change this later if you want to.